### PR TITLE
CFD-293 Homepage: Activity stream based on the filter (Logged-in users)

### PR DESCRIPTION
**Pull request by HelenaEksler**
#498

### CODE REVIEW

File: capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 1]
```diff
@@ -849,8 +849,59 @@ function c4m_message_preprocess_node__event(&$variables) {
...
+        if (!empty($interests)) {
+          $query->fieldCondition('c4m_related_topic', 'target_id', $interests, 'IN');
+        }
+        $result = $query->execute();
+        $groups += !empty($result['node']) ? array_keys($result['node']) : array();
+      }
+    }
+    else {
+      // Get all groups user has access to.
+      $query = new EntityFieldQuery();
```
Comment (amitaibu) [inline capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: this looks wrong - we can't get *all* the groups in the system

Comment (amitaibu) [inline capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: at the minimum there should be a limit

**Comment by nader77** [comment]
@HelenaEksler As you are working on the activity stream, I found a bug when the user has no groups, he will have an error when automatically refreshing the activity stream.
Use user "president".
Thanks.

**Comment by HelenaEksler** [comment]
selection_230
selection_231

**Comment by zero2one** [comment]
@HelenaEksler when I filter by "My Groups" I still see Article activity in the list.

Articles don't belong to a group and should not be listed in the stream…

c4m-activity-filter-groups

**Comment by HelenaEksler** [comment]
@zero2one It's strange - because I don't see articles when filter is set to My groups.
capacity4more

Maybe you should try to re-install.

**Comment by zero2one** [comment]
@HelenaEksler 

I got also a fatal error in the theming layer when I:

* Log in as "president"
* Set his interests to "earth"
* Go to the home page
* filter by interests => Fatal error.

I'm rebuilding the platform to be sure that nothing went wrong…
I'll keep you posted

**Comment by HelenaEksler** [comment]
@zero2one  Same thing here - don't get any errors
selection_232

selection_233

**Comment by zero2one** [comment]
@HelenaEksler After re-install:
* Filter by my groups works
* No PHP errors.

**Comment by zero2one** [comment]
QC OK > Merged.

**Pull request merged**